huma_test.go

```diff
@@ -45,6 +45,14 @@ type UUID struct {
 	uuid.UUID
 }
 
+// Node is a custom type for testing recursive definition for huma.Register
+type Node struct {
+	Name  string          `json:"name"`
+	Nodes []Node          `json:"nodes,omitempty"`
+	Left  *Node           `json:"left,omitempty"`
+	Named map[string]Node `json:"named,omitempty"`
+}
```

Comment:
I know you took this from your example case! But what about using more explicit names like `RecusiveEmbedded`, following existing test type names?